### PR TITLE
DEV: Convert install theme modal to GJS

### PR DESCRIPTION
**Pull request by martin-brennan**
We will likely get rid of this at some point, but in the meantime
let's make it easier to modify and bring it up to date.

### CODE REVIEW

File: app/assets/javascripts/admin/addon/components/modal/install-theme.gjs [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+            <div class="install-theme-warning">
+              ⚠️
+              {{this.themeCannotBeInstalled}}
+            </div>
+          {{/if}}
+          <DButton
+            @action={{this.installTheme}}
+            @disabled={{this.installDisabled}}
+            class="btn
+              {{if this.themeCannotBeInstalled 'btn-danger' 'btn-primary'}}"
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/components/modal/install-theme.gjs:460]:
```suggestion
            class={{if this.themeCannotBeInstalled "btn-danger" "btn-primary"}}
```

File: app/assets/javascripts/admin/addon/components/modal/install-theme.gjs [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+                {{/if}}
+                {{#if this.showPublicKey}}
+                  <div class="public-key">
+                    <div class="label">
+                      {{i18n "admin.customize.theme.public_key"}}
+                    </div>
+                    <div class="public-key-text-wrapper">
+                      <textarea
+                        class="public-key-value"
+                        readonly={{true}}
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/components/modal/install-theme.gjs:390]:
```suggestion
                        readonly="true"
```

File: app/assets/javascripts/admin/addon/components/modal/install-theme.gjs [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+  keyGenUrl = this.args.model.keyGenUrl || "/admin/themes/generate_key_pair";
+  importUrl = this.args.model.importUrl || "/admin/themes/import";
+
+  get createTypes() {
+    return [
+      { name: I18n.t("admin.customize.theme.theme"), value: THEMES },
+      { name: I18n.t("admin.customize.theme.component"), value: COMPONENTS },
+    ];
+  }
+
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/components/modal/install-theme.gjs:52]: I think you can extract it to a const

Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/components/modal/install-theme.gjs:52]: Yep fair enough

**Review by CvX — APPROVED** [review]

**Pull request merged**